dmi-id-dbus-service/src/main.rs

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use clap::Parser;
+use procfs::cmdline;
+use std::fs;
+use uuid::Uuid;
+use zbus::{dbus_interface, ConnectionBuilder};
+
+const DMI_SERIAL_FILE_PATH: &str = "/sys/class/dmi/id/board_serial";
+const DEFAULT_NAMESPACE: &str = "f79ad91f-c638-4889-ae74-9d001a3b4cf8";
+
+#[derive(Debug, Parser)]
+struct Cli {
+    // Retrieve hardware id from file
+    #[clap(short, long, conflicts_with_all=&["seco-dmi-serial","kernel-cmdline-key"])]
+    file_path: Option<String>,
+
+    // Shortcut per file-path = "/sys/class/dmi/id/board_serial"
+    #[clap(short, long, takes_value = false, conflicts_with_all=&["file-path","kernel-cmdline-key"])]
+    seco_dmi_serial: bool,
```

Comment:
This is not specific to SECO boards, so let's rename it to just `dmi_serial`. Also since it is a boolean a meaningful verb should be used, like `use`, therefore: `use_dmi_serial` might be an option.